flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamIterationTail.java

```diff
@@ -64,6 +70,22 @@ public void init() throws Exception {
 		super.init();
 	}
 
+	@Override
+	protected boolean performCheckpoint(CheckpointMetaData checkpointMetaData, CheckpointOptions checkpointOptions, CheckpointMetrics checkpointMetrics) throws Exception {
+		LOG.debug("Starting checkpoint {} on task {}", checkpointMetaData.getCheckpointId(), getName());
+
+		synchronized (getCheckpointLock()) {
+			if (isRunning()) {
+				dataChannel.put(new Either.Right(new CheckpointBarrier(checkpointMetaData.getCheckpointId(), checkpointMetaData.getTimestamp(), checkpointOptions)));
+				getEnvironment().acknowledgeCheckpoint(checkpointMetaData.getCheckpointId(), checkpointMetrics);
```

Comment:
Hm, not really. I cannot think of a possible usage of loaded operators in IterationSink tasks in the future too.